pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QuerySchedulerFactory.java

```diff
@@ -89,12 +96,16 @@ public static QueryScheduler create(PinotConfiguration schedulerConfig, QueryExe
 
   @Nullable
   private static QueryScheduler getQuerySchedulerByClassName(String className, PinotConfiguration schedulerConfig,
-      QueryExecutor queryExecutor) {
+      QueryExecutor queryExecutor, ServerMetrics serverMetrics, LongAccumulator latestQueryTime) {
     try {
-      Constructor<?> constructor =
-          Class.forName(className).getDeclaredConstructor(Configuration.class, QueryExecutor.class);
+      Constructor<?> constructor = Class.forName(className)
```

Comment:
Should we not be using PluginManager for class loads?

Reply:
Good point. Done